lit/docs/guides/pipeline-guides.lit

```diff
@@ -0,0 +1,676 @@
+    put steps. It's likely that a \reference{put-step} only needs a subset of
+    all available artifacts generated throughout the job.
+
+    There are two ways to specify which artifacts to send to a
+    \reference{put-step}. You can specify \code{detect} as the
+    \reference{schema.step.put-step.inputs} or you can pass in an exact list of
+    all artifacts the \reference{put-step} needs.
+
+    Using \code{detect}:
+    \codeblock{yaml}{{
+    jobs:
+    - name: the-job
+      plan: # Get some artifacts
+      - get: apples
+      - get: apple-basket
+      - get: monkeys
+      # using detect will result in "apples-location" and "basket" being passed in
+      # "monkeys" will not be passed in
+      - put: apples-in-basket
+        input: detect
+        params:
+          apples-location: apples/location # matches the first get step
+          basket: apple-basket # matches the second get step
+    }}
+
+    Specifying the exact inputs needed for the \reference{put-step}:
+    \codeblock{yaml}{{
+    jobs:
+    - name: the-job
+      plan: # Get some artifacts
+      - get: apples
+      - get: apple-basket
+      - get: monkeys
+      - put: apples-in-basket
+        input: [apples, apple-basket] # specify the exact inputs needed
+        params:
+          apples-location: apples/location
+          basket: apple-basket
+    }}
+
+  }
+
+  \section{
+    \title{Putting Task Config's in Files}
+
+    A lot of the pipeline examples that you will find on this site and in
+    resource repos will embed a \reference{schema.step.task-step.config} directly
+    in the pipeline. This is a nice way of clearly seeing what inputs/outputs the
+    task uses. Tasks are usually designed to be used in multiple places, maybe
+    with slightly different configuration. To support this scenario, most users
+    store task configs in files instead of embedding the config directly in the
+    pipeline.
+
+    Here's what this looks like in practice:
+
+    \titled-codeblock{todays-date.yml, stored in some git repo under a folder named "task"}{yaml}{{
+    platform: linux
+
+    image_resource: # define a default image for the task to use
+      type: registry-image
+      source:
+        repository: busybox
+
+    run:
+      path: date
+      args: ["+%Y-%m-%d"]
+    }}
+    \inset{} {- spacer -}
+
+    Using the task in a pipeline:
+
+    \titled-codeblock{pipline.yml}{yaml}{{
+    resources:
+    - name: ci
+      type: git
+      source:
+        uri: http://github.com/username/my-ci-repo
+
+    jobs:
+    - name: the-best-job
+      plan:
+      - get: ci
+      - task: what-day-is-it
+        file: ci/tasks/todays-date.yml
+    }}
+  }
+
+  \section{
+    \title{\code{Get} Images for Tasks Instead of using Anonymous Image Resources}
+
+    It is easy to let Concourse fetch images for tasks right when they are
+    needed, it's the easy out-of-the-box solution. Another way is to pass the
+    image(s) for tasks as inputs to jobs.
```

Comment:
Maybe we could touch on *why* you might want to do this (for instance, you want to trigger a job on new versions of an image)